### PR TITLE
[Theming] added support for theming

### PR DESCRIPTION
**Pull request by svenefftinge**
A singleton ThemingService allows to register multiple themes and switch between them.
We cannot wait for injection, which is why I made it a global.

Signed-off-by: Sven Efftinge <sven.efftinge@typefox.io>

### CODE REVIEW

**Comment by svenefftinge** [comment]
The light theme needs #866 to look pretty.

**Comment by svenefftinge** [comment]
theming

**Comment by svenefftinge** [comment]
Fixes #298

File: packages/core/src/browser/theming.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    editorTheme: 'vs',
+    activate() {
+        light.use();
+    },
+    deactivate() {
+        light.unuse();
+    }
+};
+
+export class ThemeService {
```
Comment (akosyakov) [inline packages/core/src/browser/theming.ts]: Could it be provided via the DI? And also `darkTheme` and `lightTheme` configured in the DI context?

File: packages/monaco/src/browser/monaco-editor-provider.ts [hunk 2]
```diff
@@ -21,12 +21,17 @@ import { MonacoEditorService } from './monaco-editor-service';
...
 
-const monacoTheme = 'vs-dark';
-monaco.editor.setTheme(monacoTheme);
-document.body.classList.add(monacoTheme);
+function changeTheme(editorTheme: string | undefined) {
+    const monacoTheme = editorTheme || 'vs-dark';
+    monaco.editor.setTheme(monacoTheme);
+    document.body.classList.add(monacoTheme);
+}
+changeTheme(ThemeService.get().getCurrentTheme().editorTheme);
```
Comment (akosyakov) [inline packages/monaco/src/browser/monaco-editor-provider.ts]: Could not it be done in the application contribution for the Monaco extension in `initialize` or  `initializeLayout`?

File: packages/monaco/src/browser/monaco-editor-provider.ts [hunk 3]
```diff
@@ -21,12 +21,17 @@ import { MonacoEditorService } from './monaco-editor-service';
...
+ThemeService.get().themeChange.event(event => changeTheme(event.newTheme.editorTheme));
```
Comment (akosyakov) [inline packages/monaco/src/browser/monaco-editor-provider.ts]: i think `ThemeService.themeChange` is expose ability to fire events to clients

File: packages/core/src/browser/theming.ts [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+    private themes: { [id: string]: Theme } = {};
+    private activeTheme: Theme | undefined;
+    themeChange = new Emitter<ThemeChangeEvent>();
```
Comment (akosyakov) [inline packages/core/src/browser/theming.ts]: You can do:
```ts
protected readonly _themeChange = new Emitter<ThemeChangeEvent>();
public readonly themeChange = _themeChange.event;
```

File: packages/core/src/browser/theming.ts [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    }
+
+    onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): void {
+        const themeService = ThemeService.get();
+        const items = themeService.getThemes().map(t =>
+            new QuickOpenItem({
+                label: t.label,
+                description: t.description,
+                run(mode: QuickOpenMode) {
+                    themeService.setCurrentTheme(t.id);
```
Comment (akosyakov) [inline packages/core/src/browser/theming.ts]: I wonder if I preview one theme and then close the quick panel will it rollback the current theme?

File: packages/core/src/browser/theming.ts (hunk 1 above)
Comment (svenefftinge) [reply packages/core/src/browser/theming.ts]: no, it is too late. we want to set the theme before everything.

File: packages/monaco/src/browser/monaco-editor-provider.ts (hunk 2 above)
Comment (svenefftinge) [reply packages/monaco/src/browser/monaco-editor-provider.ts]: see above

File: packages/monaco/src/browser/monaco-editor-provider.ts (hunk 3 above)
Comment (akosyakov) [reply packages/monaco/src/browser/monaco-editor-provider.ts]: ok

File: packages/monaco/src/browser/monaco-editor-provider.ts (hunk 2 above)
Comment (akosyakov) [reply packages/monaco/src/browser/monaco-editor-provider.ts]: ok

File: packages/core/src/browser/theming.ts (hunk 5 above)
Comment (svenefftinge) [reply packages/core/src/browser/theming.ts]: good catch. fixed it

File: packages/core/src/browser/theming.ts (hunk 4 above)
Comment (svenefftinge) [reply packages/core/src/browser/theming.ts]: ok

**Pull request merged**